google-cloud-contrib/google-cloud-nio/src/test/java/com/google/cloud/storage/contrib/nio/it/ITGcsNio.java

```diff
@@ -354,6 +357,45 @@ public void testListFiles() throws IOException {
     }
   }
 
+
+  @Test
+  public void testDeleteRecursive() throws IOException {
+    try (FileSystem fs = getTestBucket()) {
+      List<Path> paths = new ArrayList<>();
+      paths.add(fs.getPath("Racine"));
+      paths.add(fs.getPath("playwrights/Moliere"));
+      paths.add(fs.getPath("playwrights/French/Corneille"));
+      for (Path path : paths) {
+        Files.write(path, FILE_CONTENTS, UTF_8);
+      }
+      deleteRecursive(fs.getPath("playwrights/"));
+      assertThat(Files.exists(fs.getPath("playwrights/Moliere"))).isFalse();
+      assertThat(Files.exists(fs.getPath("playwrights/French/Corneille"))).isFalse();
+      assertThat(Files.exists(fs.getPath("Racine"))).isTrue();
+      Files.deleteIfExists(fs.getPath("Racine"));
+    }
```

Comment:
Maybe add at the end:

`assertThat(Files.exists(fs.getPath("Racine"))).isFalse();`